### PR TITLE
Add license check to submissions

### PR DESCRIPTION
**Pull request by Quozlet**
Currently only for GitHub submissions

### CODE REVIEW

**Comment by Quozlet** [comment]
I haven't testing this as I just don't have all the config needed to actually start up the bot. I did test the GraphQL query though: https://developer.github.com/v4/explorer/

Response for a valid license is the SPDX id, otherwise the license field is null

**Comment by Quozlet** [comment]
```js
const Discord = require('discord.js');
const { GraphQLClient, gql } = require('graphql-request');
const client = new Discord.Client();
const ghClient = new GraphQLClient('https://api.github.com/graphql')
ghClient.setHeader('Authorization', `bearer ${process.env.GITHUB_TOKEN}`)
client.once('ready', () => {
    console.log('Ready!');
});

client.on('message', async message => {
    if (!message.content.startsWith('?') || message.author.bot) return;
    try {
        const url = new URL(message.content.slice('?'.length).trim().split(' ').shift());
        const ghQuery = gql`
        query ($url: URI!) {
          resource(url: $url) {
            ... on Repository {
              licenseInfo {
                spdxId
              }
            }
          }
        }  
        `
        const licenseData = await ghClient.request(ghQuery, { url: url.toString() })
        message.channel.send(`Eligible? ${!!url.hostname.contains("github.com")} Valid license? ${JSON.stringify(licenseData)} says ${!!(licenseData?.resource?.licenseInfo)}`);
    } catch (e) {
        message.channel.send(`Error: ${e}`)
    }

});

client.login(process.env.DISCORD_TOKEN);
```
ss

File: src/db/index.ts [hunk 1]
```diff
@@ -175,3 +179,23 @@ export async function adjustDownvotesForProject (type: 'add' | 'remove', id: Dis
     return { success: true, wasRejected: hasEnoughDownvotes, wasPaused: project.paused, reason: '', project }
   }
 }
+
+export function isEligibleForLicenseCheck (submission: ProjectSubmission): boolean {
+  return !!submission.links.source.includes('github.com')
```
Comment (linuswillner) [inline src/db/index.ts:184]: Unnecessary boolean coercion
```suggestion
  return submission.links.source.includes('github.com')
```

File: src/db/index.ts [hunk 2]
```diff
@@ -175,3 +179,23 @@ export async function adjustDownvotesForProject (type: 'add' | 'remove', id: Dis
...
+      ... on Repository {
+        licenseInfo {
+          spdxId
+        }
+      }
+    }
+  }
+  `
+  const licenseData = await ghClient.request(ghQuery, { url: submission.links.source })
+  return !!(licenseData as GitHubLicenseData)?.resource?.licenseInfo
```
Comment (linuswillner) [inline src/db/index.ts:200]: Would prefer a bit more block segmentation here for the sake of readability
```suggestion
  `
  
  const licenseData = await ghClient.request(ghQuery, { url: submission.links.source })
  
  return !!(licenseData as GitHubLicenseData)?.resource?.licenseInfo
```

**Review by linuswillner — CHANGES_REQUESTED** [review]
A few nits small nits here and there, and then there's a big one.

Why is the license checking logic in the DB controller? To me, it seems entirely unrelated to anything the `nedb` database does. Please split it into a separate file, perhaps something like `utils/licenseCheck.ts` would be apt.

Otherwise stellar work. 👍🏻

**Comment by linuswillner** [comment]
This seems otherwise good now, package.json apparently changed under your feet though. If you could just fix that, I'd be happy to merge this 👍🏻

**Review by linuswillner — APPROVED** [review]

**Pull request merged**